### DISCUSSION TITLE
Advertise as HomePod

### DISCUSSION THREAD

**Discussion by MichaelTJohnston** [mikebrady/shairport-sync#1588] [Q&A]
Hi there,

Is there any way to advertise the shairport-sync AirPlay 2 service to look as though it's a HomePod? I would like to use shairport as my Apple TV's "Default Audio Output" but I believe that only works with HomePods. I'm open to other suggestions to achieve this goal as well.

Thanks!

**Comment by ghost** [comment]
I can tell you that there have been rumors of this feature coming later to non-Apple devices.

**Comment by mikebrady** [comment]
Thanks for the suggestion!

**Comment by nicopasla** [comment]
Would love to have that, the default option always reset on Apple tv

**Comment by julian-baumann** [comment]
Are there any news to this issue? Is it possible to fake a homepod?

**Reply by mikebrady** [reply]
Don't think so -- there's a lot more to a HomePod than Shairport Sync can provide...

**Reply by mansidak** [reply]
@mikebrady @julian-baumann I was able to simulate a HomePod earlier today by setting the "am" value to 'AudioAccessory5,1'. Btw, was anyone able to compile this project on an iOS device? I'm curious how can I convert my iOS device into an airplay server...

https://github.com/mikebrady/shairport-sync/discussions/1935


EDIT: https://openairplay.github.io/airplay-spec/service_discovery.html#subtype

**Reply by mikebrady** [reply]
Thanks. @mansidak, I don’t know if that’s enough to allow Shairport Sync to masquerade as a HomePod.

**Reply by julian-baumann** [reply]
I only need the Apple TV to recognize Shairplay as a HomePod, so I can set it as primary audio output. Do you think that works, by only setting this value? I don't need any other HomePod functionality

**Reply by mansidak** [reply]
@mikebrady , @julian-baumann I'm able to connect and send audio to it (AirAura in this case) just fine so I wanna say it's working (??). I'm not sure what other configuration would be necessary. Attaching a screenshot.

Btw, @mikebrady, not sure if you saw my other posts was curious if you know if anyone had any luck running shairport on an iPhone -- effectively turning your iPhone into an airplay receiver?

I'm asking this since although I was able to convert my iPhone into an airplay receiver using the classis Shairplay, it doesn't support multi-room audio. I noticed that shairport sync now supports Airplay 2 and so do iPhones now so I figured I'd ask for some references. Much appreciated 🙇‍♂️

![image](https://github.com/user-attachments/assets/905041eb-85b5-42d5-a31f-38afe53ce613)

**Comment by mikebrady** [comment]
Thanks for this.

> @mikebrady , @julian-baumann I'm able to connect and send audio to it (AirAura in this case) just fine so I wanna say it's working (??). I'm not sure what other configuration would be necessary. Attaching a screenshot.

The [question](https://github.com/mikebrady/shairport-sync/discussions/1588#discussion-4621034) is whether it can be used a a Default Audio Output device or not.

> Btw, @mikebrady, not sure if you saw my other posts was curious if you know if anyone had any luck running shairport on an iPhone -- effectively turning your iPhone into an airplay receiver?

Haven't tried this, but I guess the answer is no. AirPlay 2 operation requires [exclusive access to ports 319 and 320](https://github.com/mikebrady/shairport-sync/blob/master/AIRPLAY2.md#general), and I guess they are in use with iOS.

**Reply by mansidak** [reply]
@mikebrady, I'm currently looking at how Airstream might've done it. I'm not sure if it's using Airplay 1 or 2 but it's using the shairplay

https://github.com/qasim/Airstream

**Reply by julian-baumann** [reply]
@mansidak, how did you manage to advertise Shairport Sync as a HomePod? I’ve updated all the `am=…` entries in both `rtsp.c` and `mdns.h` to `am=AudioAccessory5,1` then compiled and installed it, but it still only appears as a regular speaker.

**Edit**: 
I got it. It shows the HomePod icon on my Apple TV and iPhone, but as soon as I connect to it, it changes the icon to the default speaker icon. And it is still not possible to use it as primary output device on the Apple TV.

**Reply by nicopasla** [reply]
It’s possible to set any AirPlay receiver as default audio in tvOS 26 coming in September !
